src/rpc/server/scala/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala

```diff
@@ -85,7 +85,7 @@ class EditorService(implicit val system: ActorSystem) extends Editor {
   ): Future[CreateViewportResponse] =
     (editors ? SessionOp(
       in.sessionId,
-      View(in.offset, in.capacity, in.viewportIdDesired, in.eventInterest)
+      View(in.offset, in.capacity, in.isFloating, in.viewportIdDesired, in.eventInterest)
```

Comment:
Looking really good.  I think around line 214 of this file you'll need to populate the offset which can change if the viewport is floating.

Reply:
Fixed this made the `viewportEventKind` always be `omega_edit.ViewportEventKind.VIEWPORT_EVT_UPDATED` since with viewport subscribe it would always be just updated correct?

Reply:
No, that event gets fired if the viewport _itself_ gets modified.

Reply:
Oh but wouldn't a change in the viewport in turn update the viewport? Or is that preferred `ViewportEventKind` that should used during this method as it doesn't seem like there is a dynamic way to get the kind of the `ViewportEvent`

Reply:
VIEWPORT_EVT_UPDATED is probably a poor choice of names.  I think changing it to VIEWPORT_EVT_MODIFIED would be a better choice.  I'll add that to the backlog for later on.

I think we'll need to do it properly and use a mapping function like the C++ RPC server does to map Ωedit library events to their RPC equivalent events.  There are session events also that would benefit from such a mapping function for session events.

Reply:
@Shanedell, there isn't a super clean _dynamic_ way to do the mapping (the events in Ωedit are powers of 2 for masking and the protocol buffer event enumerations are incremental).  The log of the Ωedit event _might_ map to the incremental event IDs in protocol buffers, but that can be fragile.  I think it's better to have the explicit map, but the log method _may_ work.

Reply:
@Shanedell, the numeric values of the enumerations in the Ωedit library and in the protocol buffers match up.

Reply:
@scholarsmate When you say the enumerations is that a specific variable you are referencing?